### PR TITLE
monkey patch upstream dns to rewrite .loki cname answers

### PR DESCRIPTION
**Pull request by majestrate**

### CODE REVIEW

File: llarp/handlers/tun.cpp [hunk 1]
```diff
@@ -380,12 +380,54 @@ namespace llarp
     {
       // llarp::LogInfo("Tun.HandleHookedDNSMessage ", msg.questions[0].qname, "
       // of type", msg.questions[0].qtype);
+      std::string qname;
+      if(msg.answers.size() > 0)
+      {
+        const auto &answer = msg.answers[0];
+        if(answer.HasCNameForTLD(".loki"))
```
Comment (jagerman) [inline llarp/handlers/tun.cpp]: Shouldn't this have a `and not (has A record)`?

Comment (majestrate) [reply llarp/handlers/tun.cpp]: it will strip any A records out

**Comment by jagerman** [comment]
Also should handle .snode

File: llarp/dns/message.cpp [hunk 2]
```diff
@@ -120,36 +121,32 @@ namespace llarp
...
-        if(!ar.Decode(buf))
+        if(!rr.Decode(buf))
         {
-          llarp::LogError("failed to decode additonal");
+          llarp::LogError("failed to decode additional");
           return false;
         }
-        llarp::LogDebug(ar);
       }
+      */
```
Comment (jagerman) [inline llarp/dns/message.cpp]: I feel like this commented-out chunk of code is going to bit rot until we forget why it's commented out.

Comment (majestrate) [reply llarp/dns/message.cpp]: all lost, like tears in rain.

**Pull request merged**